ext/updater.go

```diff
@@ -163,24 +168,31 @@ func (u *Updater) StartPolling(b *gotgbot.Bot, opts *PollingOpts) error {
 	}
 
 	go u.Dispatcher.Start(b, bData.updateChan)
-	go u.pollingLoop(bData, reqOpts, v)
+	go u.pollingLoop(opts.ParentCtx, bData, reqOpts, v)
 
 	return nil
 }
 
-func (u *Updater) pollingLoop(bData *botData, opts *gotgbot.RequestOpts, v map[string]string) {
+func (u *Updater) pollingLoop(ctx context.Context, bData *botData, opts *gotgbot.RequestOpts, v map[string]string) {
 	bData.updateWriterControl.Add(1)
 	defer bData.updateWriterControl.Done()
 
 	for {
+		select {
+		case <-bData.stopUpdates:
```

Comment:
This duplicates the check in `shouldStopUpdates`

Reply:
I don't think we need this; RequestWithContext would handle the exit for us if necessary